File: Swift/Kotlin/NumberService.kt
class NumberService {

  // Funkcja nie powinna dla żadnej wartości inta zwrócić napisu
  // "That should not happen at all"
  // napisz testy automatyczne w klasie NumberServiceTests które znajdą błąd
  public fun IsItTheAnswerToUltimateQuestionOfLife(number: Int): String {
    return when (number) {
      in 0 until 10, in 90 until 100 -> "Definitely no!"
      in 10 until 20, in 80 until 90 -> "Rather no"
      in 20 until 30, in 60 until 80 -> "Well no..."
      in 30 until 40, in 50 until 60 -> "No but close..."
      in 40 until 42, in 42 until 50 -> "Almost there!"
      42 -> "Exactly!"
      else -> "That should not happen at all"
    }
  }
}

File: Swift/Services/NumberService.swift
class NumberService {
  
  public func IsItTheAnswerToUltimateQuestionOfLife(number: Int) -> String {
    switch number {
      case let n where n < 0 || n > 100:
      return "Definitely no!"
      case let n where (n > 0 && n < 10) || (n > 90):
      return "Rather no"
      case let n where n >= 10 && n < 20 || (n > 80):
      return "Well no..."
      case let n where n >= 20 && n < 30 || (n > 60):
      return "No but close..."
      case let n where n >= 30 && n < 40 || (n > 50):
      return "No but close..."
      case let n where (n >= 40 && n < 42) || (n > 42 && n < 50):
      return "Almost there!"
      case 42:
      return "Exactly!"
      default:
      return "That should not happen at all"
    }
  }
  public func IsPrime(i: Int) -> Bool {
    if i == 1 {
        return false
    }
    throw NSError(domain: "Please create a test first", code: 0, userInfo: nil)
  }
}

